extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log/vpc_flow_log_unmarshaler.go

```diff
@@ -0,0 +1,365 @@
+	case "account-id":
+		key.accountID = value
+	case "vpc-id":
+		record.Attributes().PutStr("aws.vpc.id", value)
+	case "subnet-id":
+		record.Attributes().PutStr("aws.vpc.subnet.id", value)
+	case "instance-id":
+		record.Attributes().PutStr(conventions.AttributeHostID, value)
+	case "az-id":
+		record.Attributes().PutStr("aws.az.id", value)
+	case "interface-id":
+		record.Attributes().PutStr("aws.eni.id", value)
+	case "srcport":
+		if err := addNumber(field, value, conventions.AttributeSourcePort); err != nil {
+			return false, err
+		}
+	case "dstport":
+		if err := addNumber(field, value, conventions.AttributeDestinationPort); err != nil {
+			return false, err
+		}
+	case "protocol":
+		n, err := getNumber(value)
+		if err != nil {
+			return false, err
+		}
+		protocolNumber := int(n)
+		if protocolNumber < 0 || protocolNumber >= len(protocolNames) {
+			return false, fmt.Errorf("protocol number %d does not have a protocol name", protocolNumber)
+		}
+		record.Attributes().PutStr(conventions.AttributeNetworkProtocolName, protocolNames[protocolNumber])
+	case "type":
+		record.Attributes().PutStr(conventions.AttributeNetworkType, strings.ToLower(value))
+	case "region":
+		key.region = value
+	case "flow-direction":
+		switch value {
+		case "ingress":
+			record.Attributes().PutStr(conventions.AttributeNetworkIoDirection, "receive")
+		case "egress":
+			record.Attributes().PutStr(conventions.AttributeNetworkIoDirection, "transmit")
+		default:
+			return true, fmt.Errorf("value %s not valid for field %s", value, field)
+		}
+	case "version":
+		if err := addNumber(field, value, "aws.vpc.flow.log.version"); err != nil {
+			return false, err
+		}
+	case "srcaddr":
+		// If pkt-srcaddr exists, then this might be an intermediary address.
+		// Otherwise, place it in source.address.
+		_, found := record.Attributes().Get(conventions.AttributeSourceAddress)
+		if found {
+			record.Attributes().PutStr("source.layer.address", value)
+		} else {
+			record.Attributes().PutStr(conventions.AttributeSourceAddress, value)
+		}
+	case "pkt-srcaddr":
+		// contrary to srcaddr logic
+		if intermediary, found := record.Attributes().Get(conventions.AttributeSourceAddress); found {
+			record.Attributes().PutStr("source.layer.address", intermediary.Str())
+		}
+		record.Attributes().PutStr(conventions.AttributeSourceAddress, value)
+	case "dstaddr":
+		// same as srcaddr
+		_, found := record.Attributes().Get(conventions.AttributeDestinationAddress)
+		if found {
+			record.Attributes().PutStr("destination.layer.address", value)
+		} else {
+			record.Attributes().PutStr(conventions.AttributeDestinationAddress, value)
+		}
+	case "pkt-dstaddr":
+		// same logic as pkt-srcaddr
+		if intermediary, found := record.Attributes().Get(conventions.AttributeSourceAddress); found {
+			record.Attributes().PutStr("destination.layer.address", intermediary.Str())
+		}
+		record.Attributes().PutStr(conventions.AttributeDestinationAddress, value)
```

Comment:
Might be simpler to collect these particular fields and set the attributes after processing all fields.

Reply:
Otherwise I _think_ the logic is OK, though I'm having trouble following because of all the if/elses and my brain is a bit fried.

I'm not a fan of the `foo.layer.address` names. I'd like to get to a conclusion on the other thread about whether there's ever a case where the two pairs are both different in one record.

Reply:
Resolving. Issue in discussion at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38897#discussion_r2015979155.